spec/services/media_storage/azure_adapter_spec.rb

```diff
@@ -0,0 +1,172 @@
+require 'spec_helper'
+
+RSpec.describe MediaStorage::AzureAdapter do
+  let(:storage_account_name) { 'tiny-watermelons' }
+  let(:container) { 'test' }
+  let(:opts) do
+    {
+      azure_storage_account: storage_account_name,
+      azure_storage_access_key: 'fake',
+      azure_storage_container: container,
+      stub_responses: true
```

Comment:
Does this actually work in the azure landscape? in the aws client this avoids any communication with the remote cloud storage service.

Reply:
You know what, it looks like it doesn't. I definitely overlooked this, when I was first checking out the aws adapter specs I thought it was a built in rspec tool but then it slipped my mind to go back and check whether it worked as I expected.

When myself and @zwolf were working on tove, we had discussed using the [VCR](https://github.com/vcr/vcr) gem (though not sure if we actually ended up using it or not, since I'm not seeing it in the Gemfile), I'm thinking that would work well for this type of scenario, i.e. recording responses from http requests and saving them so that next time that request is made, it'll just use the saved response rather than making the request again.

Will put it on the to-do list.

Reply:
I'd be in favour of using verifying doubles for the azure method calls and trust the gem does what we want (based on the rails azure adapter example we used as a template). 

I would strongly vote against having this rely on external services and adding credentials against this sever for testing etc.

Reply:
Extremely agree about keeping tests independent of external services. These specs, though, seem to just be using the gem to build and format the various URLs by feeding it fake credentials and testing the output. I do also agree though that this could be done with verifying doubles and a bit of trust that the input will generate the output without actually testing the gem.

Reply:
That is, test that the gem is receiving the method calls that you expect, with the arguments that you expect, and leave it at that rather than letting the gem do the processing and testing the output. 

VCR is useful for testing your own code that relies on a API response, but in this case we already have the API requests/responses abstracted away by the gem. We can just test that we're giving the gem the right arguments to the right method calls.

Reply:
Azure client and storage objects have been replaced with verifying doubles 👍